### PR TITLE
wip: Exit out when failed to download kic image and fallback images.

### LINKED ISSUE #8997: failed to download kic base image or any fallback image (unable to access gcr.io)
~ minikube start 

Pulling base image ...
E0814 15:09:05.833724    5268 cache.go:175] Error downloading kic artifacts:  failed to download kic base image or any fallback image
🔥  Creating docker container (CPUs=2, Memory=6144MB) ...
🤦  StartHost failed, but will try again: creating host: create: creating: setting up container node: preparing volume for minikube container: docker run --rm --entrypoint /usr/bin/test -v minikube:/var gcr.io/k8s-minikube/kicbase:v0.0.11@sha256:6fee59db7d67ed8ae6835e4bcb02f32056dc95f11cb369c51e352b62dd198aa0 -d /var/lib: exit status 125

~ docker pull gcr.io/k8s-minikube/kicbase:v0.0.11

Error response from daemon: Get https://gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

i can pull other images, just this failed

### PR DESCRIPTION
**Pull request by vishjain**
fixes #8997

### CODE REVIEW

File: pkg/minikube/node/cache.go [hunk 1]
```diff
@@ -173,7 +173,9 @@ func beginDownloadKicBaseImage(g *errgroup.Group, cc *config.ClusterConfig, down
 					return nil
 				}
 			}
-			klog.Infof("failed to download %s, will try fallback image if available: %v", img, err)
+			infoStr := fmt.Sprintf("failed to download %s, will try fallback image if available: %v", img, err)
+			image.ErrFailedDownloadKICImages = errors.New(infoStr)
```
Comment (ilya-zuyev) [inline pkg/minikube/node/cache.go:177]: Usually, global `Err*` variables are supposed to be read-only, excluding maybe init functions. The idea is to be able to have checks like `if err == image.ErrFailedDownloadKICImages { ... `.  Changing value of `Err*` makes such checks undetermined and hard to debug.

**Review by ilya-zuyev — COMMENTED** [review]
Could you add PR description attaching minikube logs before and after the changes?

File: pkg/minikube/image/types.go [hunk 2]
```diff
@@ -21,3 +21,6 @@ var ErrNeedsLogin error
 
 // ErrGithubNeedsLogin is thrown when user needs to login specifically to github packages)
 var ErrGithubNeedsLogin error
+
+// ErrFailedDownloadKICImages is thrown when user
```
Comment (sharifelgamal) [inline pkg/minikube/image/types.go:25]: can you finish this comment?

**Comment by vishjain** [comment]
> Could you add PR description attaching minikube logs before and after the changes?

It's a bit hard to do that

File: pkg/minikube/node/cache.go [hunk 3]
```diff
@@ -193,6 +195,8 @@ func waitDownloadKicBaseImage(g *errgroup.Group) {
 			}
 			if errors.Is(err, image.ErrGithubNeedsLogin) || errors.Is(err, image.ErrNeedsLogin) {
 				exit.Message(reason.Usage, `Please either authenticate to the registry or use --base-image flag to use a different registry.`)
+			} else if errors.Is(err, image.ErrFailedDownloadKICImages) {
+				exit.Message(reason.Usage, "Failed to download KIC base image or any fallback image.")
```
Comment (medyagh) [inline pkg/minikube/node/cache.go:199]: this is not a "Usage" problem !

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by medyagh** [comment]
@vishjain still working on this ?

**Comment by vishjain** [comment]
@medyagh Making changes, Thanks!

**Pull request closed without merging**